src/apps/advanced-search/AdvancedSearchRow.tsx

```diff
@@ -66,16 +67,26 @@ const AdvancedSearchRow: React.FC<AdvancedSearchRowProps> = ({
     newData.rows[newData.rows.length - 1].id =
       newData.rows[newData.rows.length - 2].id + 1;
     updateData(newData);
+    // Update the focus.
+    setFocusedRow(rowIndex + 1);
   };
-
   const removeRow = (
     index: number,
     updateData: (data: AdvancedSearchQuery) => void
   ) => {
     const newData = { ...data };
     newData.rows.splice(index, 1);
     updateData(newData);
+    // Update the focus. If we're removing the first row, focus the new first row.
+    setFocusedRow(index === 0 ? 0 : index);
```

Comment:
Could this be simplified to:
```suggestion
    setFocusedRow(index);
```